src/svg.draw.js

```diff
@@ -1,248 +1,242 @@
+  point (event) {
+    if (this.point !== this.start) {
+      return this.start(event)
+    }
+    if (this.pointPlugin) {
+      return this.pointPlugin(event)
+    }
+
+    // If this function is not overwritten we just call stop
+    this.stop(event)
+  }
+
+  // The stop-function does the cleanup work
+  stop (event) {
+    if (event) {
+      this.update(event)
+    }
+
+    // Plugin may want to clean something
+    if (this.clean) { this.clean() }
+
+    // Unbind from all events
+    off(window, 'mousemove.draw')
+    this.parent.off('click.draw')
+
+    // remove Refernce to PaintHandler
+    this.el.forget('_paintHandler')
+
+    // overwrite draw-function since we never need it again for this element
+    this.el.draw = function () {
+    }
+
+    // Fire the `drawstop`-event
+    this.el.fire('drawstop')
+  }
+
+  // Updates the element while moving the cursor
+  update (event) {
+
+    if (!event && this.lastUpdateCall) {
+      event = this.lastUpdateCall
+    }
+
+    this.lastUpdateCall = event
+
+    // Get the current transform matrix
+    // it could have been changed since the start or the last update call
+    this.m = this.el.node.getScreenCTM().inverse()
+
+    // Call the calc-function which calculates the new position and size
+    this.calc(event)
+
+    // Fire the `drawupdate`-event
+    this.el.fire('drawupdate', { event: event, p: this.p, m: this.m })
+  }
+
+  // Called from outside. Finishs a poly-element
+  done () {
+    this.calc()
+    this.stop()
+
+    this.el.fire('drawdone')
+  }
+
+  // Called from outside. Cancels a poly-element
+  cancel () {
+    // stop drawing and remove the element
+    this.stop()
+    this.el.remove()
+
+    this.el.fire('drawcancel')
+  }
+
+  // Calculate the corrected position when using `snapToGrid`
+  snapToGrid (draw) {
+
+    var temp = null
+
+    // An array was given. Loop through every element
+    if (draw.length) {
+      temp = [draw[0] % this.options.snapToGrid, draw[1] % this.options.snapToGrid]
+      draw[0] -= temp[0] < this.options.snapToGrid / 2 ? temp[0] : temp[0] - this.options.snapToGrid
+      draw[1] -= temp[1] < this.options.snapToGrid / 2 ? temp[1] : temp[1] - this.options.snapToGrid
+      return draw
+    }
+
+    // Properties of element were given. Snap them all
+    for (var i in draw) {
+      temp = draw[i] % this.options.snapToGrid
+      draw[i] -= (temp < this.options.snapToGrid / 2 ? temp : temp - this.options.snapToGrid) + (temp < 0 ? this.options.snapToGrid : 0)
+    }
+
+    return draw
+  }
+
+  param (key, value) {
+    this.options[key] = value === null ? defaults[key] : value
+    this.update()
+  }
+
+  // Returns the plugin
+  getPlugin () {
+    return drawPlugins[this.el.type]
+  }
+}
+
+export function registerPlugin (name, obj) {
+  var plugins = {}
+  if (typeof name === 'string') {
+    plugins[name] = obj
+  } else {
+    plugins = name
+  }
+
+  for (var shapes in plugins) {
+    var shapesArr = shapes.trim().split(/\s+/)
```

Comment:
For splitting you can use the regex `delimiter` which is exported with the `regex` object (`import {regex} from ...`)